apps/system/js/base_module.js

```diff
@@ -107,24 +139,48 @@
   var SubmoduleMixin = {
     loadWhenIdle: function(modules) {
       return this.service.request('schedule', () => {
+        var submodules = this._convertModulesToArray(modules || []);
         this.constructor.SUB_MODULES =
-          this.constructor.SUB_MODULES.concat(modules);
+          this.constructor.SUB_MODULES.concat(submodules);
         return this._startSubModules();
       });
     },
+
+    _convertModulesToArray: function(submodules) {
+      var modules = [];
+      var defaulModules;
+      var deviceModules;
+
+      if (Array.isArray(submodules)) {
+        modules = submodules;
+      } else {
+        defaulModules = submodules.default || [];
+        deviceModules = submodules[DEVICE_TYPE] || [];
+        modules = defaulModules.concat(deviceModules);
+      }
+
+      return modules;
+    },
+
     /**
      * Helper function to load and start the submodules defined in
      * |this.constructor.SUB_MODULES|.
      */
     _startSubModules: function() {
-      if (!this.constructor.SUB_MODULES ||
-          this.constructor.SUB_MODULES.length === 0) {
+      var submodules = this.constructor.SUB_MODULES;
+      if (!submodules ||
+          (Array.isArray(submodules) && submodules.length === 0) ||
+          (!Array.isArray(submodules) &&
+            (!submodules.default || submodules.default.length === 0) &&
+            (!submodules[DEVICE_TYPE] || submodules[DEVICE_TYPE].length === 0)))
+      {
         return Promise.resolve();
       }
 
-      var submodules = this.constructor.SUB_MODULES.slice();
+      this.constructor.SUB_MODULES = this._convertModulesToArray(submodules);
```

Comment:
Sorry, please ignore the above comment. I didn't noticed the `SUB_MODULES` has already been modified in `loadWhenIdle` method.